memcopy.h

```diff
@@ -18,9 +18,19 @@ static inline uint64_t load_64_bits(const unsigned char *in, unsigned bits) {
   MEMCPY(&chunk, in, sizeof(chunk));
 
 #if BYTE_ORDER == LITTLE_ENDIAN
-    return chunk << bits;
+  return chunk << bits;
+#elif defined(HAVE_BUILTIN_BSWAP64)
+  return __builtin_bswap64(chunk) << bits;
 #else
-    return __builtin_bswap64(chunk) << bits;
+  chunk = ((chunk & 0xFF00000000000000u) >> 56u) |
```

Comment:
For clarity and easier re-use of code, perhaps we should make this a macro instead? I don't have any personal experience with coding for big endian systems, so is it likely that we will need this elsewhere at some point?
If it is not likely to be needed elsewhere, I'd still prefer to have this as a macro that checks for __builtin_bswap64 and provides it if it is not provided by the compiler. Hmm, that would also avoid the whole need for checks in cmake/configure I think. Thoughts?

Reply:
When I did the patch for big-endian PPC systems, I had all byteswap macros grouped in one header file. This made it pretty easy to cleanup some places where data was stored byte by byte instead of byte swapping and then using single store for each variable or struct member.

Reply:
I found out that zutil.h already had a `ZSWAP32` macro.  I amended the previous patch by removing the configure/cmake detection and providing the `ZSWAP64` macro defined under the same logic as `ZSWAP32`.

Reply:
I separately pushed the ZSWAP64 change in #232 as we will also need a ZSWAP16.